### PR TITLE
feat(widgets): add autocomplete

### PR DESCRIPTION
**Pull request by Haroenv**
This is simply an empty widget, to be used with third-party components.

Todo:
- [x] docs
- [x] tests
- [x] make example work (cc @darrenjennings)

[Story with autosuggest](https://deploy-preview-512--vue-instantsearch.netlify.com/stories/?selectedKind=AutoComplete&selectedStory=using%20vue-autosuggest&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs)

### CODE REVIEW

File: stories/Autocomplete.stories.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        this.selected = selected;
+      },
+      indicesToSuggestions(indices) {
+        return indices.map(({ hits }) => ({
+          data: hits.map(
+            hit =>
+              hit.name
+                ? hit
+                : // the ATIS index doesn't use `name`, so we make it pretend it does
+                  Object.assign(hit, {
```
Comment (francoischalifour) [inline stories/Autocomplete.stories.js]: Better not mutate `hit`: `Object.assign({}, hit, { ... })`

File: src/components/Autocomplete.vue [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    };
+  },
+  beforeCreate() {
+    this.connector = connectAutocomplete;
+  },
+  computed: {
+    widgetParams() {
+      return {
+        indices: this.indices,
+        escapeHits: this.escapeHTML,
```
Comment (francoischalifour) [inline src/components/Autocomplete.vue]: Aren't we calling that `escapeHTML` from now on?

File: package.json [hunk 3]
```diff
@@ -83,6 +83,7 @@
     "rollup-plugin-vue": "3.0.0",
     "rollup-watch": "4.3.1",
     "vue": "2.5.17",
+    "vue-autosuggest": "^1.5.0",
```
Comment (francoischalifour) [inline package.json]: Dependency should be exact.

File: src/components/Autocomplete.vue (hunk 2 above)
Comment (Haroenv) [reply src/components/Autocomplete.vue]: Exactly, that's what it's called here, `escapeHits` is the connector's name, `escapeHTML` the Vue one

File: package.json (hunk 3 above)
Comment (Haroenv) [reply package.json]: Will change those in another PR, they are behind in most :)

File: package.json [hunk 4]
```diff
@@ -83,7 +83,7 @@
     "rollup-plugin-vue": "3.0.0",
     "rollup-watch": "4.3.1",
     "vue": "2.5.17",
-    "vue-autosuggest": "^1.5.0",
+    "vue-autosuggest": "1.5.0",
```
Comment (darrenjennings) [inline package.json]: Btw 1.6 is out now, maybe not relevant here https://github.com/Educents/vue-autosuggest/releases/tag/v1.6.0

**Comment by Haroenv** [comment]
oh oops, I didn't see that was a comment, not ✅ @francoischalifour

**Pull request merged**